internal/services/network/public_ip_prefix_resource.go

```diff
@@ -246,7 +265,9 @@ func resourcePublicIpPrefixRead(d *pluginsdk.ResourceData, meta interface{}) err
 	if props := resp.PublicIPPrefixPropertiesFormat; props != nil {
 		d.Set("prefix_length", props.PrefixLength)
 		d.Set("ip_prefix", props.IPPrefix)
-
+		if props.CustomIPPrefix != nil && props.CustomIPPrefix.ID != nil {
+			d.Set("custom_ip_prefix_id", props.CustomIPPrefix.ID)
+		}
```

Comment:
both this field and `ip_version` need to have a value set into the state, these should be:

```suggestion
		customIPPrefixID := ""
		if props.CustomIPPrefix != nil && props.CustomIPPrefix.ID != nil {
			customIPPrefixID = *props.CustomIPPrefix.ID // optionally rewriting this by parsing this with an ID Formatter if required
		}
		d.Set("custom_ip_prefix_id", customIPPrefixID)
```